sklearn/preprocessing/_label.py

```diff
@@ -647,33 +730,52 @@ def _inverse_binarize_multiclass(y, classes):
 
         return classes[y_i_argmax]
     else:
-        return classes.take(y.argmax(axis=1), mode="clip")
+        xp, _, device_ = get_namespace_and_device(y, xp=xp)
+        classes = xp.asarray(classes, device=device_)
+        indices = xp.argmax(y, axis=1)
+        indices = xp.clip(indices, 0, classes.shape[0] - 1)
 
+        return classes[indices]
 
-def _inverse_binarize_thresholding(y, output_type, classes, threshold):
+
+def _inverse_binarize_thresholding(y, output_type, classes, threshold, xp=None):
     """Inverse label binarization transformation using thresholding."""
 
     if output_type == "binary" and y.ndim == 2 and y.shape[1] > 2:
         raise ValueError("output_type='binary', but y.shape = {0}".format(y.shape))
 
-    if output_type != "binary" and y.shape[1] != len(classes):
+    xp, _, device_ = get_namespace_and_device(y, xp=xp)
+    classes = xp.asarray(classes, device=device_)
```

Comment:
I think classes will be on the same `namespace` and `device` as `y` because this function is only called from the `LabelBinarizer` class. Or is there some case where this is not true.

Reply:
I think then it makes sense to keep it this way.